src/librustdoc/formats/cache.rs

```diff
@@ -131,44 +130,21 @@ struct CacheBuilder<'a, 'tcx> {
 }
 
 impl Cache {
-    crate fn from_krate<'tcx>(
-        render_info: RenderInfo,
-        document_private: bool,
+    crate fn new(access_levels: AccessLevels<DefId>, document_private: bool) -> Self {
+        Cache { access_levels, document_private, ..Cache::default() }
+    }
+
+    crate fn populate(
+        &mut self,
+        mut krate: clean::Crate,
+        tcx: TyCtxt<'_>,
         extern_html_root_urls: &BTreeMap<String, String>,
         dst: &Path,
-        mut krate: clean::Crate,
-        tcx: TyCtxt<'tcx>,
-    ) -> (clean::Crate, Cache) {
+    ) -> clean::Crate {
```

Comment:
Feels like it might be worth adding a doc-comment along the lines of:

```markdown
Populates the `Cache` with more data. The returned `Crate` will be missing some data
that was in `krate` due to the data being moved into the `Cache`.
```

Reply:
Looks like you added it 👍